apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/JvmTaskConfigurator.kt

```diff
@@ -2,6 +2,7 @@ package com.apollographql.apollo.gradle.internal
 
 import org.gradle.api.NamedDomainObjectContainer
 import org.gradle.api.Project
+import org.gradle.api.internal.jvm.ClassDirectoryBinaryNamingScheme
```

Comment:
Hmm, I'm not sure about using internal APIs. This bite me back a lot of times before. Task names are public. Gradle and Kotlin team will never change task names like compileKotlin or compileFooJava. But they may do internal refactoring which even removed/renames this class.

Little duplication of code where you need to remove `Main` is not too bad.

Reply:
Shoots, I didn't see this was internal. You're completely right, that'll break if/when gradle decides to move things around. I'll try to duplicate the logic here. It's not only about the "Main" thing but also about `-` and maybe some other characters ?

Reply:
It should be better now. The java plugin uses a regex to find word boundaries, which explains why `-` was considered a regex.